### DISCUSSION TITLE
Border offset is wrong when using colorful border or text

### DISCUSSION THREAD

**Discussion by GlebBeloded** [charmbracelet/lipgloss#332] [Q&A] [answered]
For some reason when I edit the color of a style and add a border, border lines seem to be misaligned. At first I looked at https://github.com/charmbracelet/lipgloss/issues/55 for solution.
I've tried different emulators and fonts to see what was causing the problem, but wasn't able to figure it out by myself.

Here is the code:
```go
func main() {
	t := lipgloss.JoinVertical(lipgloss.Left,
		lipgloss.NewStyle().Border(lipgloss.ThickBorder()).BorderForeground(lipgloss.Color("9")).Render("Hello, world!"),
		lipgloss.NewStyle().Border(lipgloss.ThickBorder()).Foreground(lipgloss.Color("2")).Render("Hello, world!"),
		lipgloss.NewStyle().Border(lipgloss.ThickBorder()).Render("Hello, world!"),
		lipgloss.NewStyle().Border(lipgloss.ThickBorder()).Render("Hello, world!"),
	)

	t = lipgloss.JoinHorizontal(lipgloss.Top, t, t, t)
	fmt.Println(t)
	//root.Execute()
}
```

and this is how it looks in alacritty: 
<img width="778" alt="image" src="https://github.com/user-attachments/assets/360d4162-4a13-4848-8841-c39221e45668">

For some reason emptyspace is added before the border when it or the text inside contains color
Thank you for your help in advance!

**Comment by GlebBeloded** [comment, marked as answer]
I've figured out by debug that the issue was in ansi.StringWidth function.
Going to the library github, I've found people who found a way to fix it.
https://github.com/charmbracelet/x/issues/123

All you need to do is to downgrade:
```go
replace github.com/charmbracelet/x/ansi v0.1.3 => github.com/charmbracelet/x/ansi v0.1.2
```
which results in expected output: 
<img width="739" alt="image" src="https://github.com/user-attachments/assets/7be8b402-963c-4ef5-9cb6-ae02ba5a6a9d">

**Comment by shaunco** [comment]
This was fixed by https://github.com/charmbracelet/x/pull/125 for `github.com/charmbracelet/x/ansi v0.1.4`

**Reply by meowgorithm** [reply]
Thanks for your help with this one! We referenced your posts in our fixes.
